spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitAdmin.java

```diff
@@ -523,8 +526,22 @@ private void declareExchanges(final Channel channel, final Exchange... exchanges
 
 			if (!isDeclaringDefaultExchange(exchange)) {
 				try {
-					channel.exchangeDeclare(exchange.getName(), exchange.getType(), exchange.isDurable(),
-						exchange.isAutoDelete(), exchange.getArguments());
+					if (exchange.isDelayed()) {
+						Map<String, Object> arguments = exchange.getArguments();
+						if (arguments == null) {
+							arguments = new HashMap<String, Object>();
+						}
+						else {
+							arguments = new HashMap<String, Object>(arguments);
```

Comment:
I was thinking it would be better to not change the user's arguments map. I added a boolean to represent this argument instead.